pkg/build/debian.go

```diff
@@ -27,9 +28,18 @@ func Debian(manifest model.Manifest) error {
 	if err := util.RunMake(manifest, "istio", nil, "deb/fpm"); err != nil {
 		return fmt.Errorf("failed to build sidecar.deb: %v", err)
 	}
-	if err := util.CopyFile(path.Join(manifest.RepoOutDir("istio"), "istio-sidecar.deb"), path.Join(manifest.OutDir(), "deb", "istio-sidecar.deb")); err != nil {
-		return fmt.Errorf("failed to package istio-sidecar.deb: %v", err)
+	if util.FileExists(path.Join(manifest.RepoOutDir("istio"), "istio-sidecar.deb")) {
+		if err := util.CopyFile(path.Join(manifest.RepoOutDir("istio"), "istio-sidecar.deb"), path.Join(manifest.OutDir(), "deb", "istio-sidecar.deb")); err != nil {
+			return fmt.Errorf("failed to package istio-sidecar.deb: %v", err)
+		}
+	} else if util.FileExists(path.Join(os.Getenv("TARGET_OUT_LINUX"), "release", "istio-sidecar.deb")) {
```

Comment:
Where is TARGET_OUT_LINUX coming from? That variable is set per repo, so I am a bit confused why it would be set in release-builders environment?

Reply:
When built in a container, the files are built into /work/out/ (ex: /work/out/linux_amd64/release/istio-sidecar.deb). When built using prow, there are built in the tmp directories. We either need to check both places in the copy, or do as I did for archives, and always force the built artifacts to go to the /tmp directories.

Reply:
Also, it appears in the logs that we build the .deb and .rpm in a few places. Before and after the images are built. It's the later copy that is failing.

Reply:
The docker portion of the build depends on the .deb/.rpm so they get built as part of that process. We build them again in the deb and rpm portions of the build and copy the files in the later portions.

Reply:
See prior remark about unsetting env vars.